### DISCUSSION TITLE
IIS Reverse proxy

### DISCUSSION THREAD

**Discussion by adamjehu** [Budibase/budibase#6376] [Help]
Hey,
I have a self-hosted instance which I would like to have exposed via IIS reverse proxy on another machine using ARR. The reverse proxy works well to login to Budibase and to build apps. But once I try to preview any of my apps, or launch it when published, I get an Error 500. If the app is launched on my local network and localhost, there is no problem, so I'm thinking there must be an issue with the reverse proxy. Is there any additional configurations which I may have missed? Apologies for how vague my issue is; let me know what information I could provide which may help.

**Comment by shogunpurple** [comment]
Hi @adamjehu - I do not know a lot about IIS and application request routing, but I think the best place to start here is to see what 500 error you get in the logs when you try to preview.

If you are just getting "Internal Server Error" as the message on the 500, it would be good to look at the logs when you try and preview your budibase application.

`docker-compose logs --follow` then try to see if a specific error is thrown when you preview. If so, can you share that error here and we can help you debug it. 

Thanks!

**Reply by adamjehu** [reply]
Hello @shogunpurple - thank you getting back to me so quickly. The recurring error I am getting is:

budibase-couchdb-service-1 | [notice] 2022-05-26T06:27:19.920468Z couchdb@127.0.0.1 <0.322.0> -------- chttpd_auth_cache changes listener died because the _users database does not exist. Create the database to silence this notice.
budibase-couchdb-service-1 | [error] 2022-05-26T06:27:19.920762Z couchdb@127.0.0.1 emulator -------- Error in process <0.14185.4> on node 'couchdb@127.0.0.1' with exit value:
budibase-couchdb-service-1 | {database_does_not_exist,[{mem3_shards,load_shards_from_db,"_users",[{file,"src/mem3_shards.erl"},{line,399}]},{mem3_shards,load_shards_from_disk,1,[{file,"src/mem3_shards.erl"},{line,374}]},{mem3_shards,load_shards_from_disk,2,[{file,"src/mem3_shards.erl"},{line,403}]},{mem3_shards,for_docid,3,[{file,"src/mem3_shards.erl"},{line,96}]},{fabric_doc_open,go,3,[{file,"src/fabric_doc_open.erl"},{line,39}]},{chttpd_auth_cache,ensure_auth_ddoc_exists,2,[{file,"src/chttpd_auth_cache.erl"},{line,198}]},{chttpd_auth_cache,listen_for_changes,1,[{file,"src/chttpd_auth_cache.erl"},{line,145}]}]}

Would this be the cause or should I look a little deeper?

Also, bbworker throws this:
    }
    err: {
      "type": "ForbiddenError",
      "message": "Unauthorized - no public worker access",
      "stack":
          ForbiddenError: Unauthorized - no public worker access
              at Object.throw (/app/node_modules/koa/lib/context.js:97:11)
              at /app/dist/api/index.js:109:18
              at dispatch (/app/node_modules/koa-compose/index.js:42:32)
              at /app/node_modules/@koa/router/lib/router.js:368:16
              at dispatch (/app/node_modules/koa-compose/index.js:42:32)
              at /app/node_modules/@budibase/pro/dist/index.js:1:316908
              at Generator.next (<anonymous>)
              at /app/node_modules/@budibase/pro/dist/index.js:1:3817
              at new Promise (<anonymous>)
              at j (/app/node_modules/@budibase/pro/dist/index.js:1:3562)
              at /app/node_modules/@budibase/pro/dist/index.js:1:316704
              at dispatch (/app/node_modules/koa-compose/index.js:42:32)
              at /app/node_modules/@koa/router/lib/router.js:368:16
              at dispatch (/app/node_modules/koa-compose/index.js:42:32)
              at /app/node_modules/@budibase/backend-core/dist/src/middleware/csrf.js:51:20
              at Generator.next (<anonymous>)
      "status": 403,
      "statusCode": 403,
      "expose": true
    }

**Reply by adamjehu** [reply]
Hi @shogunpurple 
So I was able to replicate the error without using the reverse proxy - where I tried to access an app using its published URL: localserver:10000/app/appname without being logged in; using my browser's incognito mode.

In the past, I would have been sent to the login screen, and once successful then be redirected to the app I am trying to access. Instead I was presented with:

{"message":"Missing/invalid DB name","status":500}

The problem I think is not with the reverse proxy, but with redirected authentication - sorry for the now misleading subject of this thread. I'm not sure if this assumption is correct, let me know if it makes sense given the logs I sent and if I can provide any more information to help.

**Reply by adamjehu** [reply]
Hi @shogunpurple, I deleted the Budibase container and pulled the latest, and after upgrading my app to v1.0.214, I no longer get the {"message":"Missing/invalid DB name","status":500} error mentioned in my last message when accessing it locally. The error 500 however still occurs when trying to access Budibase through the IIS reverse proxy. When a request is made this way, bbaps give this in the log:

INFO  [1657043394077] (30 on 7e4af2172445): request completed
    res: {
      "statusCode": 200,
      "headers": {
        "vary": "Accept-Encoding",
        "content-length": "787",
        "last-modified": "Thu, 30 Jun 2022 12:25:07 GMT",
        "cache-control": "max-age=0",
        "content-type": "text/html; charset=utf-8"
      }
    }
    responseTime: 12
    req: {
      "id": 455,
      "method": "GET",
      "url": "/builder",
      "headers": {
        "connection": "upgrade",
        "x-real-ip": "172.18.0.1",
        "x-forwarded-for": "<externalip>:23162, 172.18.0.1",
        "host": "app-service:4002",
        "cache-control": "max-age=0",
        "accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9",
        "accept-encoding": "gzip, deflate, br",
        "accept-language": "en-US,en;q=0.9",
        "max-forwards": "10",
        "user-agent": "Mozilla/5.0 (Linux; Android 12) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.5060.70 Mobile Safari/537.36",
        "upgrade-insecure-requests": "1",
        "sec-gpc": "1",
        "sec-fetch-site": "none",
        "sec-fetch-mode": "navigate",
        "sec-fetch-user": "?1",
        "sec-fetch-dest": "document",
        "x-original-url": "/builder",
        "x-arr-ssl": "2048|256|C=US, O=Let's Encrypt, CN=R3|CN=<mydomain.com>",
        "x-arr-log-id": "6ebd1569-ee59-46c3-b022-d85357e783e6"
      },
      "remoteAddress": "::ffff:172.18.0.9",
      "remotePort": 35866
    }

Any thoughts on what could be the cause?
